src/SVDiagnosticCollection.ts

```diff
@@ -340,14 +340,20 @@ export class SVDiagnosticCollection implements Disposable {
         for (const [key, unmappedDiagnostics] of this.unmappedIssuesCollection.entries()) {
             const indexOfUnmappedDiagnostic: number = unmappedDiagnostics.indexOf(diagnostic);
             if (indexOfUnmappedDiagnostic >= 0) {
+                // If we have a large number of diagnostics, deleting the key
+                // and resetting it is far better than having two allocations going at once.
```

Comment:
What would be the "two allocations going at once"?

Reply:
Yeah, there isn't one. I've removed this comment and the delete.

Also, this code had an issue with the way it was remapping. You cannot "splice" and iterate async at the same time (that results in strange behavior obviously).

We now keep track of "what" needs to be moved between the collections and do that after automatic remapping has completed.

---
In reply to: [406540906](https://github.com/microsoft/sarif-vscode-extension/pull/228#discussion_r406540906) [](ancestors = 406540906)